src/main/java/com/aws/greengrass/deployment/DeploymentService.java

```diff
@@ -504,31 +504,29 @@ private void cancelCurrentDeployment() {
                     .isCancellable()) {
                 logger.atInfo().log("Deployment already finished processing or cannot be cancelled");
             } else {
-                boolean canCancelDeployment = context.get(UpdateSystemPolicyService.class).discardPendingUpdateAction(
-                        ((DefaultDeploymentTask) currentDeploymentTaskMetadata.getDeploymentTask()).getDeployment()
-                                .getGreengrassDeploymentId());
-                if (canCancelDeployment) {
-                    currentDeploymentTaskMetadata.getDeploymentResultFuture().cancel(true);
-                    DeploymentType deploymentType = currentDeploymentTaskMetadata.getDeploymentType();
-                    if (DeploymentType.SHADOW.equals(deploymentType) || DeploymentType.LOCAL.equals(deploymentType)) {
-                        deploymentStatusKeeper.persistAndPublishDeploymentStatus(
-                                currentDeploymentTaskMetadata.getDeploymentId(),
-                                currentDeploymentTaskMetadata.getGreengrassDeploymentId(),
-                                currentDeploymentTaskMetadata.getConfigurationArn(),
-                                currentDeploymentTaskMetadata.getDeploymentType(), JobStatus.CANCELED.toString(),
-                                new HashMap<>(), currentDeploymentTaskMetadata.getRootPackages());
-                    }
-                    logger.atInfo().kv(DEPLOYMENT_ID_LOG_KEY_NAME, currentDeploymentTaskMetadata.getDeploymentId())
-                            .kv(GG_DEPLOYMENT_ID_LOG_KEY_NAME,
-                                    currentDeploymentTaskMetadata.getGreengrassDeploymentId())
-                            .log("Deployment was cancelled");
-                } else {
+                boolean deploymentStillPending = context.get(UpdateSystemPolicyService.class)
```

Comment:
How is UpdateSystemPolicyService going to behave during cancellation now? Lets make sure we do not break the existing behavior as customer and internal components rely on pre and post component update events for their use cases.

Reply:
I will investigate the existing behaviors and get back to you on this

Reply:
Ran UATs, components that subscribe to update notifications receive pre component update event when deployment is being processed, and post component update event when deployment is cancelled.

Behavior looks consistent to existing behavior

Reply:
Nit: Rename `deploymentStillPending` to `canCancelDeploymentUpdateAction` or `canCancelDeploymentActivation`